### PR TITLE
fix: the first rollout of a new instance has no regression baseline

### PR DESCRIPTION
**Pull request by rbuergi**
**Measured on pearl.meshweaver.cloud, 2026-09-15/16.** A freshly provisioned portal served `503` at the edge for nine hours with a RUNNING pod:

```
Health check nodetype_bake with status Unhealthy … 'NodeType bake regressed on this image —
  refusing readiness so the rollout stalls with the previous image still serving.
  2 NodeType(s) regressed on this image: GoogleMaps/Gallery, MyAi/Panel'
```

Neither type ever built there. The plugin registry marks `GoogleMaps` and `MyAi` **preInstalled**, so their content lands on every instance, while the modules they bind (`MeshWeaver.Blazor.GoogleMaps` …) are store-delivered and had not arrived — `required_modules` said exactly that, Degraded, in the same breath.

Every entry in that instance's report was `NeverBuilt`, and `NodeTypeBakeEntry.WasHealthy` counts NeverBuilt as healthy, so a first-ever compile failure became a REGRESSION and the pod gated itself forever.

**The gate's own sentence is the argument.** It refuses readiness *"so the rollout stalls with the previous image still serving"*. On a first rollout there is no previous image and no previous pod.

**The change**
- The sweep empties the regression baseline when every entry is `NeverBuilt` — a first bake — and logs that it did.
- Narrow on purpose: one baked type means a working instance, where a new failure can be the image's doing and still gates. `NodeTypeBakeEntry.WasHealthy` is untouched.
- Failures with no baseline are no longer dropped silently: they land in `NodeTypeBakeGateState.WithoutBaseline` and are named in all three health payloads.

**Verified locally:** 10 of 10 in `MeshWeaver.Hosting.Test` for the new file plus the neighbouring gate tests; `MeshWeaver.Hosting` builds Release with `-warnaserror`, 0 warnings.

**Out of scope, worth a separate look:** the registry pre-installs content whose module the instance cannot activate, which is what put unbuildable types on a customer portal in the first place.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/MeshWeaver.Hosting/DynamicTypePreWarmer.cs [hunk 1]
```diff
@@ -805,13 +805,38 @@ private static IObservable<PreWarmOutcome> WarmPending(
...
+        // edge for hours because two types the REGISTRY pre-installs (GoogleMaps/Gallery, MyAi/Panel)
+        // could not compile — their store-delivered modules had not arrived — and every entry in the
+        // report was NeverBuilt, so each counted as "healthy before" and the pod gated itself forever.
+        //
+        // Narrow on purpose: the baseline is emptied ONLY when the report says this mesh has never
+        // built anything at all. One baked type is enough to mean a working instance, and there the
+        // gate keeps its full strictness — a newly added type that fails on an established instance
+        // still gates exactly as before, because that failure CAN be the image's doing.
+        var firstBake = report.Entries.Count > 0
+                        && report.Entries.All(e => e.State is BakeState.NeverBuilt);
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply src/MeshWeaver.Hosting/DynamicTypePreWarmer.cs]: Valid, and fixed in 17958b1ce1. The rule is now a pure pair on `DynamicTypePreWarmer` — `IsFirstBake(report)` and `RegressionBaseline(report)` — that `WarmPending` calls and the tests exercise directly, so the branch is reachable from a test.

Proven falsifiable rather than asserted: with `IsFirstBake` forced to `false`, `EveryTypeNeverBuilt_IsAFirstBake_SoTheBaselineIsEmpty` fails (1 of 7); the file was then restored byte-identical (md5) and the suite is 7/7. A second test, `OneBakedType_MeansAnEstablishedInstance_SoTheBaselineStands`, is the negative control for widening the rule.

File: src/MeshWeaver.Hosting/NodeTypeBakeGate.cs [hunk 2]
```diff
@@ -283,9 +298,18 @@ public void MarkRunning(string message)
...
             return false;
 
+        // No working build to regress FROM: pre-existing breakage, or the first bake of an instance
+        // that has never built anything. Never gates — but it is recorded, because a failure nobody
+        // can see is how a fresh portal ends up serving 503 with an empty health payload.
+        if (!outcome.WasHealthyBeforeBake)
+        {
+            withoutBaseline[outcome.TypePath] = $"{outcome.Status}: {outcome.Detail ?? "(no detail)"}";
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply src/MeshWeaver.Hosting/NodeTypeBakeGate.cs]: Valid, and the more serious of the three — fixed in 17958b1ce1. The no-baseline question is now asked LAST, after the timeout, content and retirement classifications, so each outcome keeps the bucket that says what it actually was; `withoutBaseline` holds exactly what WOULD have gated had there been something to regress from. No gating change — all four paths return `false`.

Worth stating why it was not a corner case: on a first bake EVERY outcome arrives with no baseline, so the old order would have filed every timeout, content deletion and retirement on a fresh instance as a failure it never was. Two tests pin it (`OnAFirstBake_ATimeoutIsStillNotAVerdict`, `OnAFirstBake_ARetirementIsStillNotAFailure`), and both fail if the check is moved back to the top of `MarkOutcome` — measured.

File: src/MeshWeaver.Hosting/DynamicTypePreWarmer.cs [hunk 3]
```diff
@@ -805,13 +805,38 @@ private static IObservable<PreWarmOutcome> WarmPending(
...
-        var healthyBefore = report.Entries
-            .Where(e => e.WasHealthy)
-            .Select(e => e.TypePath)
-            .ToHashSet(StringComparer.OrdinalIgnoreCase);
+        var healthyBefore = firstBake
+            ? new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+            : report.Entries
+                .Where(e => e.WasHealthy)
+                .Select(e => e.TypePath)
+                .ToHashSet(StringComparer.OrdinalIgnoreCase);
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply src/MeshWeaver.Hosting/DynamicTypePreWarmer.cs]: Agreed — fixed in 17958b1ce1. `RegressionBaseline` returns an `ImmutableHashSet<string>` on both branches (`ImmutableHashSet<string>.Empty.WithComparer(OrdinalIgnoreCase)` / `ToImmutableHashSet(OrdinalIgnoreCase)`), which is what the repository collections policy asks for.
